### DISCUSSION TITLE
Link to Uptime Monitor in Email

### DISCUSSION THREAD

**Discussion by pravyn** [jertel/elastalert2#1235] [Q&A] [answered]
Hello,

We are sending emails whenever host is down in Elasticsearch Uptime (Heartbeat). We have a need to send link to single monitor, whenever it is down. Not sure how to achieve this as each monitor url is getting dynamically constructed with some random ID at the end as sample below. Any help will be appreciated @jertel 

https://test.com:5601/app/uptime/monitor/**MTAuMC4xOC4xMDk=/**

`
alert_text: |
        <html>
        <body>
           <ul>
           <li>Host: {0}</li>
           <li>IP: {1}</li>
           <li>  <a href="https://test.com:5601/app/uptime/????????">Link to Dashboard</a> </li>
           </ul>
        </body>
        </html>`

**Comment by jertel** [comment, marked as answer]
That _random_ ID is a base64-encoded monitor ID as defined in your heartbeat.yml file. Ex:

![image](https://github.com/jertel/elastalert2/assets/16639611/881695d7-ec44-4ad3-86da-afa4273d9a89)

In your case the monitor ID is `10.0.18.109`. So if you know which monitor you want the alert to show you can base64 encode that monitor ID and add it to the URL.

**Reply by pravyn** [reply]
Thank you @jertel . I got the issue now, will do the necessary changes in heartbeat.yml and confirm you.

**Reply by pravyn** [reply]
@jertel 

How to base64 encode in heartbeat.yml ??  using processors ??

Thanks,
Pravyn

**Reply by pravyn** [reply]
@jertel 

i have used the following snippet in heartbeat.yml but iam getting some weird character's as shown below


  ```
processors:

- decode_base64_field:
      field:
        from: "monitor.id"
        to: "decoded.monitor.name"
      ignore_missing: false
      fail_on_error: true
```

![image](https://github.com/jertel/elastalert2/assets/19923046/43e81d4e-b7ef-49ce-96bf-a4788f900a57)

**Reply by jertel** [reply]
If you're trying to dynamically determine a monitor's URL then I am probably not the person to ask, as that is outside the scope of this project.

Since you're asking questions specific to the Elastic project I suggest moving the topic to the Elastic community forums, so we can keep this discussion forum focused on ElastAlert 2 software.

**Reply by pravyn** [reply]
Sure. Thanks @jertel

**Reply by Difrex** [reply]
@jertel Maybe it's can be achieved through Jinja2 filters, like in Ansible?

Something like that
```yaml
alert_text: |
  Host {{ monitor.name }} is down<br>
  <a href="https://kibana/s/all-users/app/uptime/monitor/{{ monitor.id | b64encode }}">
```

**Reply by jertel** [reply]
It's a worth a try, provided you have access to that field in your rule.
